subsys/logging/log_core.c

```diff
@@ -416,3 +416,23 @@ u32_t log_filter_get(struct log_backend const *const backend,
 		return log_compiled_level_get(src_id);
 	}
 }
+
+#ifdef CONFIG_LOG_PROCESS_THREAD
+
+#define STACKSIZE 768
+
+static void log_process_thread(void *dummy1, void *dummy2, void *dummy3)
+{
+	log_init(k_current_get());
+
+	while (1) {
+		if (log_process(false) == false) {
+			k_sleep(CONFIG_LOG_PROCESS_THREAD_SLEEP_MS);
```

Comment:
I think sleeping here is not good. We should have a way to be notified here that there are messages waiting to be processed, k_poll() could be used here for example.

Reply:
There is notifying option: number of buffered messages are tracked and when exceeds the threshold task is waken up. Sleeping is added for case when logs are not arriving continuously and some could stuck for long period. I added threshold to avoid calling k_wakeup() on every log entry because that degrades performance of the logger.

Reply:
Do you mean that this thread is put to sleep and then waken up by some other thread?

Having a k_sleep() here means that basically you are polling the logger. I think this is not really good to do it like this. Better would be that when things exceed some threshold, the logging APIs will trigger a pollable event, which causes this reading thread to be waken up and process the data. After processing this data, the thread starts to wait another events. This is a bit more code than plain sleep, but is better architecturally.

Reply:
yes, PR contains 2 commits, first commit (https://github.com/zephyrproject-rtos/zephyr/pull/8704/commits/949505964e4ad2889718724c10de162aa3a5b277) adds ability to wake up a thread when threshold is exceeded. Second commit (adding internal thread) is using this feature but additionally adds sleep period in the thread. Sleeping is added to ensure that buffered logs are processed (and presented to the developer) even when logs stopped arriving. Let say that threshold is set to 20 and you have 19 logs in a buffer and next log will not come for dozens of seconds. You may get confused in that situation if you except to see given log (e.g. you got connected, but log does not print "connected").

Sleep period is configurable (maybe i should add option where sleep period 0 sets forever sleeping).